Update to KnowRob 2.0 APi

benjaminalt opened the pull request:
Update to be compatible with the new KnowRob APIs (ask/tell replaced by kb_call/kb_project etc.)

I did not refactor all predicates in neem_interface.pl because I could not test them all. I only refactored those I used and tested.

benjaminalt committed:
Update NEEM interface to new KnowRob API

benjaminalt committed:
Update NEEM interface to new KnowRob API

benjaminalt committed:
Minimal Python NEEM interface with utilities for NEEM creation

benjaminalt committed:
neem-interface.pl: Don't assert both is_action and has_type(...)

When creating an action and specifying its type with has_type,
is_action is redundant because the fact that it is an action is
already implicit in the type. Asserting both leads to the action
individual being an individual of BOTH dul:Action and the custom action
type, which would make subsequent calls to is_individual return 2
solutions. That is not wrong, but makes querying action types
unnecessarily complex.

benjaminalt committed:
neem-interface.pl: Don't assert both is_action and has_type(...)

When creating an action and specifying its type with has_type,
is_action is redundant because the fact that it is an action is
already implicit in the type. Asserting both leads to the action
individual being an individual of BOTH dul:Action and the custom action
type, which would make subsequent calls to is_individual return 2
solutions. That is not wrong, but makes querying action types
unnecessarily complex.

benjaminalt committed:
neem_interface.py: Methods for parsing NEEMs

* Getting actions
* Getting action durations
* Getting trajectories of objects during actions

benjaminalt committed:
Wrench integration

benjaminalt committed:
Set start and end time when starting/stopping episodes

benjaminalt committed:
Better getting / setting TF

benjaminalt committed:
Adding task parameters & setting their values

benjaminalt committed:
Fix destruction error of Prolog client

An error was thrown in the destructor of the Prolog client because ros_client.terminate() was used erroneously. Remove this call.

benjaminalt committed:
Object-oriented NEEM abstraction for loading and reading from NEEMs

benjaminalt committed:
neem.py: Method to extract object trajectories

benjaminalt committed:
WIP

benjaminalt committed:
neem_interface_python: assert_transition

benjaminalt committed:
WIP for NEEM conversion from VR

Datapoint: Conversion from Unreal conventions
neem_interface.py: Added convenience methods for asserting states and
situations
neem_interface.pl: Minor tweaks

benjaminalt committed:
neem-interface.py: Replace has_type with instance_of

benjaminalt committed:
rosprolog_client: Extend ROS connection timeout

Extend the connection timeout of roslibpy from the default 10s to 20s. This was necessary in cases where ontologies/mongodb dumps took a long time to load.

benjaminalt committed:
rosprolog_client: Add throwing versions of once() and all_solutions()

Add ensure_once() and ensure_all_solutions(), which throw exceptions when the Prolog query returns False. This helps clients avoid silent failure of queries: Before, if they did not check the return value, the query could fail without the client knowing.

benjaminalt committed:
Use throwing versions of prolog.once and prolog.all_solutions

sunava commented:
Could u separate the PR, such that I can only merge the neem-interface.pl for the newer knowrob API for now?

benjaminalt committed:
Set start and end time when starting/stopping episodes

benjaminalt committed:
Better getting / setting TF

benjaminalt committed:
Remove unused URDF prefix

benjaminalt commented:
> Could u separate the PR, such that I can only merge the neem-interface.pl for the newer knowrob API for now?

Done :)

benjaminalt committed:
.gitignore: Ignore .idea and .vscode

benjaminalt committed:
Use more compact Pose representation in mem_tf_set and mem_tf_get

benjaminalt committed:
Comment out some unused predicates in neem-interface.pl

These caused errors when loading neem-interface.pl in the latest version of KnowRob, even if they were never used.

benjaminalt committed:
add_participant_with_role in neem_interface.py

benjaminalt committed:
utils.py: Proper conversion between rhs and lhs coordinate systems

benjaminalt committed:
rosprolog-client: Increase ROS connection timeout

This is necessary because sometimes it takes a very long time for the KnowRob KB / ROS to start up, which can cause timeouts.

benjaminalt committed:
Make neem_interface a proper ROS package

Add CMakeLists.txt and package.xml

benjaminalt committed:
Merge branch 'master' into vr-neem-converter

benjaminalt committed:
Minor refactorings

Add rosbridge static singleton to avoid multiple connections to ROS

benjaminalt committed:
Removed neem_interface_python

Moved neem_interface_python into own repo

benjaminalt committed:
Bugfix: mem_tf_get(Object, Pose, Timestamp) should actually take 3 args

sasjonge commented on neem-interface.pl:
```diff
@@ -2,70 +2,148 @@
+mem_add_subaction_with_task(ParentAction,SubActionType,TaskType,SubAction) :-
+    execution_agent(Agent),
+    kb_project([
+        new_iri(SubAction, SubActionType), is_individual(SubAction), instance_of(SubAction,SubActionType),
```
Hey, @NeumannM89 is looking into using the new code and we had problems with this line. What value does SubActionType have in your executions? Generally a subaction should always be a soma:'Action' or at most one of it's subclasses.

benjaminalt replied:
In my executions, SubActionType is 'http://www.ease-crc.org/ont/SOMA.owl#PhysicalAction'. Task type is something like 'http://www.ease-crc.org/ont/SOMA.owl#Grasping', and ParentAction is the IRI of the top-level action individual of the NEEM. SubAction is left uninitialized, because new_iri(SubAction, SubActionType) creates a new IRI for it.

sasjonge replied:
Ok, I was concerned, because I thought there was some confusion with Action and Task, but this is the correct usage.
